app/models/grda_warehouse/tasks/identify_duplicates.rb

```diff
@@ -112,114 +144,546 @@ def identify_duplicates
+        ),
+        client_two_name as (
+        SELECT warehouse_clients.destination_id as destination_two_id,
+          concat(regexp_replace(lower(trim(unaccent(clients."FirstName"))), '[^a-z0-9]', '', 'g'), '_',
+          regexp_replace(lower(trim(unaccent(clients."LastName"))), '[^a-z0-9]', '', 'g')) as destination_two_name
+          from #{limits}
+        )
+
+        SELECT DISTINCT
+          destination_one_id,
+          destination_two_id
+        FROM client_one_name
+        JOIN client_two_name ON (
+          -- Match on normalized name if both have it
+          destination_one_name = destination_two_name
+        )
+        WHERE destination_one_id < destination_two_id  -- Avoid duplicate pairs
+      SQL
+      # return an array of ID pairs
+      results.map { |r| [r['destination_one_id'], r['destination_two_id']] }
     end
 
-    private def find_merge_candidates
-      to_merge = Set.new
+    # Finds pairs of destination client IDs with exact date of birth matches among source clients.
+    # Uses a CTE to compare all source clients and returns pairs of destination IDs
+    # where the DOB is present, after 1920, and matches.
+    #
+    # @return [Array<Array<Integer>>] Array of [destination_one_id, destination_two_id] pairs
+    def exact_dob_matches
+      limits = <<-SQL
+        "Client" as clients
+          inner join warehouse_clients on clients.id = warehouse_clients.source_id
+          where clients."DateDeleted" is NULL
+          and clients.data_source_id != #{GrdaWarehouse::DataSource.warehouse_id} -- not a destination client
+          and clients."DOB" is not null
+          and date_part('year', clients."DOB") > 1920
+      SQL
+      results = GrdaWarehouse::Hud::Client.connection.execute(<<-SQL)
+        with client_one_dob as (
+          SELECT warehouse_clients.destination_id as destination_one_id,
+            clients."DOB" as destination_one_dob
+            from #{limits}
+        ),
+        client_two_dob as (
+          SELECT warehouse_clients.destination_id as destination_two_id,
+            clients."DOB" as destination_two_dob
+            from #{limits}
+        )
+
+        SELECT DISTINCT
+          destination_one_id,
+          destination_two_id
+        FROM client_one_dob
+        JOIN client_two_dob ON (
+          -- Match on DOB if both have it
+          destination_one_dob = destination_two_dob
+        )
+        WHERE destination_one_id < destination_two_id  -- Avoid duplicate pairs
+      SQL
+      results.map { |r| [r['destination_one_id'], r['destination_two_id']] }
+    end
+
+    # Finds pairs of destination and unprocessed source client IDs with exact SSN matches.
+    # Expands the search to any client, even those without a warehouse_client record.
+    # Only returns pairs where the SSN is valid and matches.
+    #
+    # @return [Array<Array<Integer>>] Array of [destination_client_id, source_client_id] pairs
+    def exact_ssn_matches_for_unprocessed
```

Comment:
should we use SSNDataQuality (and name/dob dq fields)? Wondering if we would want to exclude partial or incomplete data

Reply:
Historically those have been really inaccurate, but potentially we could exclude those that have indicated partial or not collected I guess.  That said, I don't know why you would have a valid SSN if the quality is not collected or partial.